c++/src/kj/string.h

```diff
@@ -296,6 +306,105 @@ class String {
   Array<char> content;
 };
 
+// =======================================================================================
+// ConstString -- A NUL-terminated Array<const char> containing UTF-8 text.
```

Comment:
Let's describe this as:

> Same as `String`, but the backing buffer is const. This has the useful property that it can reference a string literal without allocating a copy. Any `String` can also convert (by move) to `ConstString`, transferring ownership of the buffer.